### ISSUE TITLE
Consider accepting whitespaces in Template literals/Template strings in jsx

### ISSUE THREAD

**Issue by kud** [eslint/eslint#9575]
I've got this with whitespaces

```
more={`Texte N° ${object._embedded.latestText.reference}`}
```

<img width="596" alt="image" src="https://user-images.githubusercontent.com/655838/32373484-c58e5338-c099-11e7-970c-7ce2a4a352dd.png">

but eslint returns me: `error  Irregular whitespace not allowed  no-irregular-whitespace`

which is sad.

Could you consider accepting whitespaces there?

I read this https://eslint.org/docs/rules/no-irregular-whitespace#skiptemplates apparently you already support it but is it something linked to jsx?

My config

```javascript
module.exports = {
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "node": true,
  },

  "extends": ["eslint:recommended", "plugin:react/recommended"],

  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "allowImportExportEverywhere": true,

    // enable JSX
    "ecmaFeatures": {
      "jsx": true,
    },

  },

  "plugins": [
    "react",
  ],

  "rules": {
    "comma-dangle": ["error", "always-multiline"],
    // "indent": ["error", 2],
    "linebreak-style": ["error", "unix"],
    "no-unused-vars": ["warn"],
    "no-console": 0,
  },

  "globals": {
    "API_URL": true,
    "Raven": true,
    "analytics": true,
  }
}
```

**Comment by platinumazure** [comment]
Hi @kud, thanks for the issue!

I'm not sure I follow. According to your configuration, unless eslint:recommended or plugin:react/recommended sets `no-irregular-whitespace` with a different configuration, then your configuration is *not* using `skipTemplates: true` for this rule. The rule mentions only the `skipStrings: true` option is set by default, and eslint:recommended (at least) does not usually deviate from default settings.

You can use `eslint --print-config path/to/file.js` to see the full, effective configuration for a given rule. Hopefully this will show you if skipTemplates is configured or not.

If you do not see skipTemplates in the full config, please try adding a rule configuration for this rule:

```js
{
    // omitted
    "rules": {
        "no-irregular-whitespace": ["error", { "skipStrings": true, "skipTemplates": true }],
        // omitted
    }
}
```

**Comment by kud** [comment]
<img width="726" alt="image" src="https://user-images.githubusercontent.com/655838/32378600-74a519d6-c0ab-11e7-9a61-1fd4a0ebed0a.png">

I've indeed misread the doc, I thought it was the default for everything here. I'll add `skipTemplates` so.

Thank you very much for your time and your kind explanation @platinumazure.

**Comment by kud** [comment]
However I've put what you said,

it still renders errors.

```
  "rules": {
    "comma-dangle": ["error", "always-multiline"],
    "no-irregular-whitespace": ["error", { "skipStrings": true, "skipTemplates": true }],
    // "indent": ["error", 2],
    "linebreak-style": ["error", "unix"],
    "no-unused-vars": ["warn"],
    "no-console": "off",
  },
```

**Comment by platinumazure** [comment]
Thanks, I guess we'll need to investigate :smile:

Could you edit your post with a minimum complete example (i.e., maybe include a JSX tag around the attribute that you've posted in your example)? Basically something that would be valid JS/JSX by itself. Thanks!

**Comment by kud** [comment]
```javascript
/* eslint no-irregular-whitespace: "off" */
import React from "react"
import PropTypes from "prop-types"
import moment from "moment"

import Card from "~/components/Card"
import StepTranslator from "~/components/StepTranslator"
import WatchBillButton from "~/components/WatchBillButton"

class WatchBill extends React.PureComponent {
  static propTypes = {
    watch: PropTypes.object.isRequired,
    onChange: PropTypes.func,
  }

  static defaultProps = {
    onChange: () => {},
  }

  handleChangeWatchButton = () => {
    this.props.onChange()
  }

  render() {
    const { watch: object } = this.props

    const stepRelation =
      object._embedded.latestText.stepRelation === "produced_at"
        ? "output"
        : "input"

    return (
      <Card
        to={`/bill-${object.id}/step-${object._embedded.currentStep
          .position}/${stepRelation}/`}
        heading={object.title}
        date={moment(object._embedded.latestText.date).format("Do MMMM YYYY")}
        fate={object.accelerated ? "accelerated" : null}
        step={
          <StepTranslator
            when={object._embedded.currentStep.when}
            where={object._embedded.currentStep.where}
            who={object._embedded.currentStep.who}
            relation={stepRelation}
            short
          />
        }
        watch={
          <WatchBillButton
            id={object.id}
            onChange={this.handleChangeWatchButton}
          />
        }
        more={`Texte N° ${object._embedded.latestText.reference}`}
      />
    )
  }
}

export default WatchBill
```

**Comment by platinumazure** [comment]
I've reduced the example down to this and agree that this is reproducible:

```js
/* eslint no-irregular-whitespace: error, no-unused-vars: off */
class WatchBill {
  render() {
    return (
      <Card
        more={`Texte N° 1`}  // NOTE: Uses NBSP characters as spaces
      />
    )
  }
}

export default WatchBill
```

Which can also be verified in our [online demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWlycmVndWxhci13aGl0ZXNwYWNlOiBlcnJvciwgbm8tdW51c2VkLXZhcnM6IG9mZiAqL1xyXG5jbGFzcyBXYXRjaEJpbGwge1xyXG4gIHJlbmRlcigpIHtcclxuICAgIHJldHVybiAoXHJcbiAgICAgIDxDYXJkXHJcbiAgICAgICAgbW9yZT17YFRleHRlwqBOwrDCoDFgfSAgLy8gTk9URTogVXNlcyBOQlNQIGNoYXJhY3RlcnMgYXMgc3BhY2VzXHJcbiAgICAgIC8+XHJcbiAgICApXHJcbiAgfVxyXG59XHJcblxyXG5leHBvcnQgZGVmYXVsdCBXYXRjaEJpbGwiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjUsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc29sZSI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1rZXlzIjoyLCJuby1kdXBsaWNhdGUtY2FzZSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1lbXB0eSI6Miwibm8tZXgtYXNzaWduIjoyLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOjIsIm5vLWV4dHJhLXNlbWkiOjIsIm5vLWZhbGx0aHJvdWdoIjoyLCJuby1mdW5jLWFzc2lnbiI6Miwibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8taW5uZXItZGVjbGFyYXRpb25zIjoyLCJuby1pbnZhbGlkLXJlZ2V4cCI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZWxlc3MtZXNjYXBlIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJ1c2UtaXNuYW4iOjIsInZhbGlkLXR5cGVvZiI6Mn0sImVudiI6e319fQ==).

**Comment by kud** [comment]
Yeah sure, I'll a try. :) Thanks for your deep search.

**Comment by kud** [comment]
```
import React from "react"

class WatchAmendment extends React.PureComponent {
  render() {
    const { watch: object } = this.props

    return (
      <Card
        to={`/bill-${object._embedded.bill.id}/step-${
          object._embedded.step.position
        }/input/amendment-${object.id}/`}
        heading={`Amendement N° ${object.name}`}
        subheading={
          <span>
            <Text weight="bold" upperFirst>
              {object._embedded.bill.title}
            </Text>{" "}
            <Text weight="light">
              (Texte N° {object._embedded.text.reference})
            </Text>
          </span>
        }
        committee={
          object._embedded.committee && (
            <span>
              <FormattedMessage upperFirst id="committee" />
              {" : "}
              <Text upperFirst>{object._embedded.committee.name}</Text>
            </span>
          )
        }
        target={
          object._embedded.article
            ? `Article ${object._embedded.article.name}`
            : null
        }
        author={object.authorName}
        fate={object.fate ? object.fate.label : "submitted"}
        step={
          <StepTranslator
            when={object._embedded.step.when}
            where={object._embedded.step.where}
            who={object._embedded.step.who}
            relation="input"
            short
          />
        }
        watch={
          <WatchAmendmentButton
            id={object.id}
            onChange={this.handleChangeWatchButton}
          />
        }
      />
    )
  }
}

export default WatchAmendment
```

<img width="1242" alt="image" src="https://user-images.githubusercontent.com/655838/32716900-0b156e6c-c858-11e7-8797-9504f74c0211.png">

See? :p

**Comment by kud** [comment]
I still have the problem.
I've got this settings:

```
    "no-irregular-whitespace": [
      "warn",
      { skipStrings: true, skipTemplates: true },
    ],
```

and it stills show me a warning about this:


```
import React from "react"
import PropTypes from "prop-types"
import _ from "lodash"

import shortenArticle from "~/candies/shortenArticle"

import connect from "~/hoc/elfi"

import Text from "~/components/Text"
import FormattedMessage from "~/components/FormattedMessage"
import Card from "~/components/Card"
import StepTranslator from "~/components/StepTranslator"
import WatchAmendmentButton from "~/components/WatchAmendmentButton"

class WatchPageMainWatchAmendment extends React.PureComponent {
  static propTypes = {
    apiBills: PropTypes.object.isRequired,
    apiTexts: PropTypes.object.isRequired,
    apiCommittees: PropTypes.object.isRequired,
    apiSteps: PropTypes.object.isRequired,
    watch: PropTypes.object.isRequired,
  }

  render() {
    const {
      watch: object,
      apiBills,
      apiTexts,
      apiSteps,
      apiCommittees,
    } = this.props

    const apiBill = apiBills.getById(object._embedded.bill)
    const apiText = apiTexts.getById(object._embedded.text)
    const apiCommittee = apiCommittees.getById(object._embedded.committee)
    const apiStep = apiSteps.getById(object._embedded.step)

    const { body: bill } = apiBill.toJS()
    const { body: text } = apiText.toJS()
    const { body: committee } = apiCommittee.toJS()
    const { body: step } = apiStep.toJS()

    return (
      <Card
        to={`/bill-${bill.id}/step-${step.position}/input/amendment-${
          object.id
        }/`}
        heading={
          object._embedded.parent
            ? `Sous-Amendement N° ${object.name}`
            : `Amendement N° ${object.name}`
        }
        headingN1={
          object._embedded.parent
            ? `Amendement N° ${object._embedded.parent.name}`
            : null
        }
        subheading={
          <React.Fragment>
            <Text weight="bold" upperFirst>
              {bill.title}
            </Text>{" "}
            <Text weight="light">
              (Texte N° {text.reference})
            </Text>
          </React.Fragment>
        }
        committee={
          object._embedded.committee && (
            <React.Fragment>
              <FormattedMessage upperFirst id="committee" />
              {" : "}
              <Text upperFirst>{committee.name}</Text>
            </React.Fragment>
          )
        }
        target={shortenArticle(_.startCase(_.lowerCase(object.rawArticle)))}
        author={object.authorName}
        fate={object.fate ? object.fate.label : "filed"}
        step={
          <StepTranslator
            when={step.when}
            where={step.where}
            who={step.who}
            relation="input"
            short
          />
        }
        watch={
          <WatchAmendmentButton
            id={`bills_amendment_${object.id}`}
            objectTitle={`Amendement N° ${object.name}`}
          />
        }
      />
    )
  }
}

export default connect(
  WatchPageMainWatchAmendment,
  state => ({
    apiBills: state.apiBills,
    apiTexts: state.apiTexts,
    apiSteps: state.apiSteps,
    apiCommittees: state.apiCommittees,
  }),
)
```

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.

**Comment by kud** [comment]
Alright. Thank you. :)
